pkg/kubelet/configmap/configmap_manager.go

```diff
@@ -281,7 +281,8 @@ func (c *cachingConfigMapManager) RegisterPod(pod *v1.Pod) {
 	prev = c.registeredPods[key]
 	c.registeredPods[key] = pod
 	if prev != nil {
-		for name := range getConfigMapNames(prev) {
+		// remove entries which were in prev but are not in names
+		for name := range getConfigMapNames(prev).Difference(names) {
```

Comment:
check !names.Has to avoid reallocating?

Reply:
great catch.

+1 for Jordan`s comment.